File: backend/verification-service/src/main/java/com/certverify/verification/model/Certificate.java
package com.certverify.verification.model;

import lombok.Data;

@Data
public class Certificate {
    private String certificateId;
    private String certificateNumber;
    private String studentId;
    private String universityId;
    private String studentName;
    private String studentEmail;
    private String courseName;
    private String specialization;
    private String grade;
    private Double cgpa;
    private String issueDate;
    private String completionDate;
    private String certificateHash;
    private String digitalSignature;
    private String verificationCode;
    private String pdfPath;
    private String status;
    private String revocationReason;
    private String createdAt;
    private String updatedAt;
}

File: frontend/student-portal/src/types/certificate.ts
export interface Certificate {
  certificateId: string;
  certificateNumber: string;
  studentId: string;
  universityId: string;
  studentName: string;
  courseName: string;
  specialization?: string;
  grade: string;
  cgpa?: number;
  issueDate: string;
  completionDate: string;
  certificateHash: string;
  digitalSignature: string;
  verificationCode: string;
  pdfPath?: string;
  status: 'ACTIVE' | 'REVOKED';
  revocationReason?: string;
  createdAt: string;
  updatedAt: string;
}

export interface CertificateIssueRequest {
  studentId: string;
  universityId: string;
  studentName: string;
  courseName: string;
  specialization?: string;
  grade: string;
  cgpa?: number;
  issueDate: string;
  completionDate: string;
}

export interface CertificateUpdateRequest {
  studentName?: string;
  courseName?: string;
  specialization?: string;
  grade?: string;
  cgpa?: number;
  issueDate?: string;
  completionDate?: string;
}

export interface CertificateRevocationRequest {
  certificateId: string;
  reason: string;
}

export interface FileUploadResponse {
  fileName: string;
  filePath: string;
  fileSize: number;
  uploadTime: string;
}

export interface ApiResponse<T> {
  success: boolean;
  data?: T;
  message?: string;
  error?: string;
  timestamp: string;
}